airflow/providers/amazon/aws/transfers/ftp_to_s3.py

```diff
@@ -132,12 +132,13 @@ def execute(self, context: 'Context'):
                 if self.ftp_filenames == '*':
                     files = list_dir
                 else:
-                    files = list(filter(lambda file: self.ftp_filenames in file, list_dir))
+                    ftp_filename: str = self.ftp_filenames
+                    files = list(filter(lambda f: ftp_filename in f, list_dir))
```

Comment:
couldnt find a better way to fix this, the line 136 is fine since there is a isinstance(str) check in 125, but mypy still complains.

Reply:
Works for me.

Reply:
I think the main problem here is that `mypy` cannot reliably tell it fhe self.ftp_filenams have been overwritten by something lese. we are executing some methods on self that could potentially override it  a better solution will proably be to do this assignment before is_instance (and then you will not have to do the second check).

Reply:
Sorry confused, are you referring to moving this line
`ftp_filename: str = self.ftp_filenames`

```        if self.ftp_filenames:
            if isinstance(self.ftp_filenames, str):
                ftp_filename: str = self.ftp_filenames
                self.log.info(f'Getting files in {self.ftp_path}')
                list_dir = self.ftp_hook.list_directory(
                    path=self.ftp_path,
                )

                if self.ftp_filenames == '*':
                    files = list_dir
                else:
                    files = list(filter(lambda f: ftp_filename in f, list_dir))

                for file in files:
                    self.log.info(f'Moving file {file}')

                    if self.s3_filenames and isinstance(self.s3_filenames, str):
                        filename = file.replace(self.ftp_filenames, self.s3_filenames)```

Reply:
Ah right. My bad. Looks good :). it's s3_filenames not ftp_filenames :)